Extension/src/common.ts

```diff
@@ -727,34 +729,29 @@ export function isExecutable(file: string): Promise<boolean> {
     });
 }
 
-export function allowExecution(file: string): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
+export async function allowExecution(file: string): Promise<void> {
         if (process.platform !== 'win32') {
-            checkFileExists(file).then((exists: boolean) => {
-                if (exists) {
-                    isExecutable(file).then((isExec: boolean) => {
-                        if (isExec) {
-                            resolve();
-                        } else {
-                            fs.chmod(file, '755', (err: NodeJS.ErrnoException | null) => {
-                                if (err) {
-                                    reject(err);
-                                    return;
-                                }
-                                resolve();
-                            });
+            const exists: boolean = await checkFileExists(file);
+            if (exists) {
+                const isExec: boolean = await isExecutable(file);
+                if (isExec) {
+                    return;
+                } else {
+                    fs.chmod(file, '755', (err: NodeJS.ErrnoException | null) => {
+                        if (err) {
+                            throw new Error(err.message);
```

Comment:
`fs.chmod` is a file operation, so it may not complete within the context of this call.  So, the lambda being passed in may execute later with a different call stack.  I don't think it's safe to throw errors here.  It's similar to an event handler scenario, where you would need to continue to use resolve/reject in order to control completion of the calling function.

Or, you could switch to using `chmodSync()` to simplify the code, as a very slight impact to performance.